libass/ass_directwrite_info_template.h

```diff
@@ -12,12 +12,22 @@ static bool NAME(FONT_TYPE)(FONT_TYPE *font,
     BOOL exists;
 
     meta->weight = font->lpVtbl->GetWeight(font);
-    meta->width = map_width(font->lpVtbl->GetStretch(font));
 
     DWRITE_FONT_STYLE style = font->lpVtbl->GetStyle(font);
-    meta->slant = (style == DWRITE_FONT_STYLE_NORMAL) ? FONT_SLANT_NONE :
-                 (style == DWRITE_FONT_STYLE_OBLIQUE)? FONT_SLANT_OBLIQUE :
-                 (style == DWRITE_FONT_STYLE_ITALIC) ? FONT_SLANT_ITALIC : FONT_SLANT_NONE;
+    meta->style_flags = style == DWRITE_FONT_STYLE_NORMAL ? 0 : FT_STYLE_FLAG_ITALIC;
+
+    const char* data;
+    UINT32 size;
+    void* context;
+    hr = font->lpVtbl->TryGetFontTable(font, DWRITE_MAKE_OPENTYPE_TAG('O', 'S', '/', '2'),
+                                       &data, &size, &context, &exists);
+    if (!FAILED(hr)) {
```

Comment:
You should probably use [SUCCEEDED](https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-succeeded) instead of [!FAILED](https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-failed).